File: site/lab/solver/PatternData.js
import { find, ofArray, empty } from "./fable_modules/fable-library-js.4.16.0/Map.js";
import { comparePrimitives } from "./fable_modules/fable-library-js.4.16.0/Util.js";
import { map2 } from "./fable_modules/fable-library-js.4.16.0/Array.js";
import { ofArray as ofArray_1 } from "./fable_modules/fable-library-js.4.16.0/List.js";

let patterns = empty({
    Compare: comparePrimitives,
});

export function setData(keys, values) {
    patterns = ofArray(map2((key, rows) => [key, ofArray_1(rows)], keys, values), {
        Compare: comparePrimitives,
    });
}

export function read(key) {
    return find(key, patterns);
}

File: solver333/PatternData.fs
module PatternData

let mutable private patterns : Map<string, string list> = Map.empty

let setData (keys: string array) (values: string array array) =
    patterns <- Array.map2 (fun key rows -> key, List.ofArray rows) keys values |> Map.ofArray

let read key = Map.find key patterns
